cluster/email.go

```diff
@@ -126,20 +128,90 @@ func SendMail(toName, toEmail, subject, body string) error {
 }
 
 // GetTemplate gets a template from the templates folder and applies the template date
-func GetTemplate(templateFileName string, data interface{}) (*string, error) {
-	// Working Directory
-	wd, err := os.Getwd()
+func GetTemplate(templateName string, data interface{}) (*string, error) {
+	// try to load the template from the db
+	dbTemplate, err := getTemplateFromDB(nil, templateName)
 	if err != nil {
-		return nil, err
-	}
-	t, err := template.ParseFiles(wd + fmt.Sprintf("/cluster/templates/%s.html", templateFileName))
-	if err != nil {
-		return nil, err
-	}
+		// ignore no results error
+		if !strings.Contains(err.Error(), "no rows in result set") {
+			return nil, err
+		}
+	}
+	var t *template.Template
+	if dbTemplate != nil && *dbTemplate != "" {
+		// parse the template from the DB
+		t, err = template.New(templateName).Parse(*dbTemplate)
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		// if we found no template, load from disk
+		// Working Directory
+		wd, err := os.Getwd()
+		if err != nil {
+			return nil, err
+		}
+		t, err = template.ParseFiles(wd + fmt.Sprintf("/cluster/templates/%s.html", templateName))
```

Comment:
does the user control the templateName value in any way? what if he does `../../home/root/.ssh/id_rsa;` 

![alt text](https://media.tenor.co/images/7f6402cf7df54cdf24505ee7895326aa/raw)

We need to add some validations:

- Reject file names which contain ., .. or / (or \ under Windows)
- Limit file names to basic alphanumeric characters
- Prepend the include directory name and append the appropriate extension

Reply:
there's other files inside the container, other than the ca certs and other templates

![tenor (1)](https://user-images.githubusercontent.com/18384552/69684107-20248c80-106c-11ea-83d2-90dff9866f37.gif)